### PR TITLE
zephyr: Integrate with Zephyr's build and test systems

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added sample.yaml which is required for automatic building
mcuboot using zephyr saniticheck.

Can be run by zephyr sanitycheck locally using
--testcase-root=/path/to/mcuboot/boot/zephyr
to make it detect the sample.yaml added.

Added here as we want to keep the zephyr fork only pure mirror repo.

### CODE REVIEW

File: boot/zephyr/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+sample:
+  description: mcuboot test build sample
+  name: mcuboot
+common:
+    tags: introduction
+    harness: console
+    harness_config:
+      type: one_line
+      regex:
+        - "Hello World! (.*)"
```
Comment (mbolivar) [inline boot/zephyr/sample.yaml]: I think this part is from the hello_world sample and is not needed. The rest of the sample.yaml looks good to me. I will leave it up to you MCUboot developers if you want to have the header changes.

Comment (nvlsianpu) [reply boot/zephyr/sample.yaml]: of course - this is just a sandbox for now, will be fixed
Actually I would add at last test with configuration for serial recovery as well.

**Comment by carlescufi** [comment]
CC @nashif

File: boot/zephyr/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+sample:
+  description: mcuboot test build sample
+  name: mcuboot
+
+tests:
+  sample.bootloader.mcuboot:
+    tags: bootloader_mcuboot
+    platform_whitelist:  nrf52840_pca10056
```
Comment (carlescufi) [inline boot/zephyr/sample.yaml]: Should we build it for another platform too? I know mcuboot supports many other Zephyr boards. Perhaps start with this one and then add more later?

Comment (nvlsianpu) [reply boot/zephyr/sample.yaml]: That my thought as well.

**Comment by utzig** [comment]
What is Zephyr sanitycheck?

**Comment by carlescufi** [comment]
> What is Zephyr sanitycheck?
It's Zephyr's tool to build and run samples and tests during CI:

https://docs.zephyrproject.org/latest/guides/test/ztest.html

File: boot/zephyr/sample.yaml (hunk 2 above)
Comment (d3zd3z) [reply boot/zephyr/sample.yaml]: The frdm_k64f should always work, so maybe that one as well?

Comment (utzig) [reply boot/zephyr/sample.yaml]: I agree, `nrf52840_pca10056` and `frdm_k64f` are the two boards I use for all my Zephyr testing.

Comment (nvlsianpu) [reply boot/zephyr/sample.yaml]: Added frdm_k64f. Are you using it to test serial recovery?

**Comment by nvlsianpu** [comment]
@carlescufi  module.yaml added

File: boot/zephyr/sample.yaml (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/sample.yaml]: ^^ if yes - please give me proj.conf for it.

Comment (utzig) [reply boot/zephyr/sample.yaml]: I don't think I've used `boot_serial` with the `frdm_k64f`, but I'm gonna test it later and can send you the `proj.conf`; but it should not block this PR.

Comment (carlescufi) [reply boot/zephyr/sample.yaml]: It doesn't work, so not enabling it right now. Have disabled it now.

**Comment by carlescufi** [comment]
@mbolivar would you mind giving this a +1?

File: boot/zephyr/sample.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+sample:
+  description: mcuboot test build sample
+  name: mcuboot
+
+tests:
+  sample.bootloader.mcuboot:
+    tags: bootloader_mcuboot
+    platform_whitelist: nrf52840_pca10056 frdm_k64f
```
Comment (mbolivar) [inline boot/zephyr/sample.yaml]: The whitelist doesn't contain pca10059. Is that ok? Will it get built for this board with CDC ACM recovery enabled?

Comment (carlescufi) [reply boot/zephyr/sample.yaml]: It will get built with CDC ACM recovery, yes, I tested it locally.

**Comment by carlescufi** [comment]
This is no longer an RFC, it's ready for merging.

**Comment by carlescufi** [comment]
@utzig @d3zd3z this is ready for merging. Any objections?

File: boot/zephyr/sample.yaml (hunk 3 above)
Comment (nvlsianpu) [reply boot/zephyr/sample.yaml]: BTW - build for pca10059 without CDC ACM would be almost identical as for pca10056 (only different flash layout).

**Pull request merged**